source/rules/no-unsafe-first.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { isCallExpression, isIdentifier, typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows unsafe first/take usage in Effects and epics.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden:
+        "Unsafe `first` and `take` usage in Effects and epics are forbidden."
+    },
+    schema: [
+      {
+        properties: {
+          observable: { type: "string" }
+        },
+        type: "object"
+      }
+    ]
+  },
+  create: context => {
+    const defaultObservable = "action(s|\\$)?";
+    const invalidOperators = ["take", "first"];
+
+    const [config] = context.options;
+    const { observable = defaultObservable } = config || {};
+    const observableRegExp = new RegExp(observable);
+
+    const { couldBeObservable, isReferenceType } = typecheck(context);
+
+    return {
+      [`VariableDeclarator CallExpression:has(MemberExpression:matches([object.name=${observableRegExp}], [object.property.name=${observableRegExp}]))[callee.property.name='pipe'][arguments]:has(CallExpression[callee.name=/^(take|first)$/])`]: (
```

Comment:
Is this too much? 😐

Reply:
`VariableDeclarator` is a problem. IIRC, `redux-observable` effects are variables. However, NgRx effect won't be. The TSLint implementation does not care about that, it just looks for `pipe` calls that stem from an observable and actions-like name.

Also, I'm vary of `has` matching AST nodes that are deeper than immediate children. I'm pretty sure that has burned me in the past. If a selector-based listener is going to be used, it might need more tests to ensure that false positives are not effected.

Reply:
Also, I'm tempted to use this as an example for a talk that I'm prepping for Thursday - on why declarative abstractions are 'hard to debug'. It's a whopper of a selector!

Reply:
What can I say, I really like the selectors 😅
Funny how readable the version is now, compared to this one.
Feel free to use the whopper in your talk.